include/openrave/planner.h

```diff
@@ -507,14 +512,18 @@ class OPENRAVE_API PlannerStatus
         return statusCode&PS_HasSolution;
     }
 
-    PlannerParametersConstPtr parameters; ///< parameters used in the planner
-    std::string description;        ///< Optional, the description of how/why the error happended. Displayed to the user by the UI. It will automatically be filled with a generic message corresponding to statusCode if not provided.
-    uint32_t statusCode; // combination of PS_X fields (PlannerStatusCode)
-    IkParameterization ikparam;      // Optional,  the ik parameter that failed to find a solution.
-    std::vector<dReal> jointValues; // Optional,  the robot's joint values in rad or m
-    CollisionReportPtr report;       ///< Optional,  collision report at the time of the error. Ideally should contents contacts information.
+    PlannerParametersConstPtr parameters;   ///< parameters used in the planner
+    std::string description;                ///< Optional, the description of how/why the error happended. Displayed to the user by the UI. It will automatically be filled with a generic message corresponding to statusCode if not provided.
+    uint32_t statusCode;                    // combination of PS_X fields (PlannerStatusCode)
+    IkParameterization ikparam;             // Optional, the ik parameter that failed to find a solution.
+    // IMPORTANT: indices of vRobotJointValues and vCollisionReportString should correspond to each other!
+    std::vector<std::vector<dReal>> vRobotJointValues;  // Records robot joint values at all collisions
+    std::vector<std::string> vCollisionReportString;    // Records all collision reports
```

Comment:
Please do not rename any of the members. I don't think you have a good enough reason to change them. Only add the new members you introduced.

Reply:
Hm, all right, then.